pkg/cmd/pulumi/config/config_test.go

```diff
@@ -91,3 +91,109 @@ func TestConfigSet(t *testing.T) {
 		})
 	}
 }
+
+//nolint:paralleltest // changes global ConfigFile variable
+func TestConfigSetTypes(t *testing.T) {
+	ctx := context.Background()
+
+	cases := []struct {
+		name     string
+		args     []string
+		expected string
+		typ      string
+		path     bool
+	}{
+		{
+			name:     "toplevel bool",
+			args:     []string{"testProject:test", "true"},
+			typ:      "bool",
```

Comment:
Maybe we could duplicate the tests and have a version of each case where `typ` is `""` to capture that behaviour as well.

Reply:
Done, I added more test cases to the test above (not setting `typ` is the same as setting `typ = ""`)